channel/handlers_client.go

```diff
@@ -448,8 +448,94 @@ func (server Server) playerEmote(conn mnet.Client, reader mpacket.Reader) {
 }
 
 func (server Server) playerUseMysticDoor(conn mnet.Client, reader mpacket.Reader) {
-	// doorID := reader.ReadInt32()
-	// fromTown := reader.ReadBool()
+	plr, err := server.players.GetFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	doorOwnerID := reader.ReadInt32()
+	fromTown := reader.ReadBool()
+
+	// Find the door in the current instance
+	doorInfo, exists := plr.inst.mysticDoors[doorOwnerID]
+	if !exists {
```

Comment:
We are failing here when trying to use door. Seems like the door lookup isn't accurate?